components/shared_code/src/shared_data_model/meta/parameter.py

```diff
@@ -21,84 +23,84 @@ class ParameterType(StrEnum):
 class Parameter(NamedModel):
     """Source parameter model."""
 
-    short_name: str | None = None
-    help: str | None = Field(None, regex=r".+\.")  # noqa: A003
+    model_config = ConfigDict(validate_default=True)
+
+    short_name: str = ""
+    help: str | None = None  # noqa: A003
     help_url: HttpUrl | None = None
     type: ParameterType  # noqa: A003
     placeholder: str | None = None
     mandatory: bool = False
     default_value: str | list[str] = ""
     unit: str | None = None
-    metrics: list[str] = Field(..., min_items=1)
+    metrics: list[str] = Field(..., min_length=1)
     values: list[str] | None = None
     api_values: dict[str, str] | None = None
     validate_on: list[str] | None = None
 
-    @validator("short_name", always=True)
-    def set_short_name(cls, short_name: str | None, values) -> str | None:
+    @model_validator(mode="after")
+    def set_short_name(self) -> Self:
         """Set the short name if no value was supplied."""
-        return short_name or values["name"].lower()
+        if not self.short_name:
+            self.short_name = self.name.lower()
+        return self
 
-    @validator("help_url", always=True)
-    def check_help(cls, help_url: HttpUrl | None, values) -> HttpUrl | None:
+    @model_validator(mode="after")
+    def check_help(self) -> Self:
         """Check that not both help and help URL are set."""
-        if help_url and values.get("help"):
-            msg = f"Parameter {values['name']} has both help and help_url"
+        if self.help_url and self.help:
+            msg = f"Parameter {self.name} has both help and help_url"
+            raise ValueError(msg)
+        if self.help and not self.help.endswith("."):
+            msg = f"The help string of parameter {self.name} does not end with a dot"
```

Comment:
Aren't we adding one automatically elsewhere? Or was that for description

Reply:
Yes, for description. But with description we add a dot if the description does not end with punctuation and with the help we complain if it's missing, so it's a bit inconsistent. I'll make it more consistent.